desc/random.py

```diff
@@ -0,0 +1,198 @@
+    if isinstance(Z_scale, tuple):
+        Z_scale = (Z_scale[1] - Z_scale[0]) * rng.random() + Z_scale[0]
+    if isinstance(R0, tuple):
+        R0 = (R0[1] - R0[0]) * rng.random() + R0[0]
+
+    R_basis = DoubleFourierSeries(M=M, N=N, NFP=NFP, sym="cos" if sym else False)
+    Z_basis = DoubleFourierSeries(M=M, N=N, NFP=NFP, sym="sin" if sym else False)
+    # alpha determines how quickly amplitude decays for high M, N,
+    # normalized so that X_norm=1 for m=1
+    R_norm = np.exp(-alpha * np.sum(abs(R_basis.modes), axis=-1)) / np.exp(-alpha)
+    Z_norm = np.exp(-alpha * np.sum(abs(Z_basis.modes), axis=-1)) / np.exp(-alpha)
+
+    R_mn = R_norm * scipy.stats.truncnorm.rvs(
+        loc=1, scale=np.exp(-beta), size=R_basis.num_modes, a=-2, b=2, random_state=rng
+    )
+    Z_mn = Z_norm * scipy.stats.truncnorm.rvs(
+        loc=1, scale=np.exp(-beta), size=Z_basis.num_modes, a=-2, b=2, random_state=rng
+    )
+
+    # scale to approximate aspect ratio
+    R_scale1 = np.mean(
+        abs(R_mn)[(abs(R_basis.modes[:, 1]) == 1) & (abs(R_basis.modes[:, 2]) == 0)]
+    )
+    Z_scale1 = np.mean(
+        abs(Z_mn)[(abs(Z_basis.modes[:, 1]) == 1) & (abs(Z_basis.modes[:, 2]) == 0)]
+    )
+    R_mn *= R_scale / R_scale1
+    Z_mn *= Z_scale / Z_scale1
+    R_mn[R_basis.get_idx(0, 0, 0)] = R0
+    if not sym:
+        Z_mn[Z_basis.get_idx(0, 0, 0)] = 0  # center at Z=0
+        # flip sign and reduce magnitude of nonsymmetric modes to avoid degenerate cases
+        # with no volume. kind of ad-hoc but seems to produce reasonable results
+        R_mn[sign(R_basis.modes[:, 1]) != sign(R_basis.modes[:, 2])] *= -np.exp(-beta)
+        Z_mn[sign(Z_basis.modes[:, 1]) == sign(Z_basis.modes[:, 2])] *= -np.exp(-beta)
+
+    surf = FourierRZToroidalSurface(
+        R_mn,
+        Z_mn,
+        R_basis.modes[:, 1:],
+        Z_basis.modes[:, 1:],
+        NFP=NFP,
+        sym=sym,
+        check_orientation=False,
+    )
+    # we do this manually just to avoid the warning when creating with left handed
+    # coordinates
+    if surf._compute_orientation() == -1:
+        surf._flip_orientation()
+        assert surf._compute_orientation() == 1
+    return surf
+
+
+def random_pressure(n=8, p0=(1e3, 1e4), rng=None):
+    """Create a random monotonic pressure profile.
+
+    Profile will be a PowerSeriesProfile with even symmetry,
+    enforced to be monotonically decreasing from p0 at r=0 to 0 at r=1
+
+    Could also be used for other monotonically decreasing profiles
+    such as temperature or density.
+
+    Parameters
+    ----------
+    n : int
+        Order of polynomial.
+    p0 : float or tuple
+        Pressure on axis. If a tuple, treats as min/max for random value.
+    rng : numpy.random.Generator
+        Random number generator. If None, uses numpys default_rng
+
+    Returns
+    -------
+    pressure : PowerSeriesProfile
+        Random pressure profile.
+    """
+    assert (n // 2) == (n / 2), "n should be even"
```

Comment:
I don't think we should enforce the order to be even. You only generate random coefficients for the even powers of rho anyways, so in the case of an odd order the coefficient for the highest power can also be zero.

Reply:
The logic in the rest of the function needs the order to be even, and we need even profiles for them to be physical anyways